### PR TITLE
Feature/arcticspacefox/news cache

### PR DESCRIPTION
**Pull request by ArcticSpaceFox**
# Added news cache feature

- caching top stories
- caching each link

### CODE REVIEW

**Comment by ArcticSpaceFox** [comment]
⚠️  Still missing testing

File: index.js [hunk 1]
```diff
@@ -788,15 +791,38 @@ client.on('message', async (message) => {
...
-      data = await data.json();
+      let data = botCacheStorage.get(`news_${i}`);
+      if (!data) {
+        // construct link
+        let link = "https://hacker-news.firebaseio.com/v0/item/" + encodeURIComponent(value[i]) + ".json?print=pretty";
+        // fetch data and parse
+        data = await fetch(link);
+        data = await data.json();
+        // update the cache
+        let success = botCacheStorage.set(`news_${i}`, data, 2 * 60 * 60 * 1000);
```
Comment (ryzetech) [inline index.js:820]: timestamp formatting inconsistant to line 803

**Review by ryzetech — COMMENTED** [review]
check will be performed asap

File: index.js (hunk 1 above)
Comment (ArcticSpaceFox) [reply index.js:820]: I just set up 2 hour cache, and had it calculated like this

**Comment by ArcticSpaceFox** [comment]
Resolved :+1:

**Pull request merged**